src/Config.ts

```diff
@@ -1,32 +1,84 @@
-import { open } from "node:fs/promises";
+import { open, access, constants, FileHandle } from "node:fs/promises";
 import Logger from "./Logger.js";
 
-export default class Config {
-    public port: number = 25565;
-    public logLevel: Logger.Level = Logger.Level.INFO;
+
+export interface Config {
+    /**
+     * Port to listen on 
+    */
+    port: number;
+
+    /**
+     * The level to display logs at
+     */
+    logLevel: Logger.Level;
 
     /**
      * Kick reason for when the server is shutting down
      */
-    public shutdownKickReason: string = "Server closed";
+    shutdownKickReason: ChatComponent;
+}
 
+export class ConfigLoader {
     /**
      * Get a Config instance from a json file
      * @param file The file to read from
      * @returns a promise that resolves to a Config instance
+     * @throws if the file does not exist or is invalid
      */
     public static async fromFile(file: string): Promise<Config> {
-        let config: Config;
+        let config: Config = ConfigLoader.getDefault();
+        let logger: Logger = new Logger("Config", config.logLevel);
+        if (!(await ConfigLoader.exists(file))) {
+            logger.warn("Config does not exist, creating default '%s'", file);
+            await ConfigLoader.createDefault(file);
+            return config;
+        }
+        const fd: FileHandle = await open(file, "r");
+        const data: string = await fd.readFile("utf-8");
+        fd.close();
+
+        config = JSON.parse(data) as Config;
+
+        return config;
```

Comment:
```suggestion
        if (!(await ConfigLoader.exists(file))) {
            await ConfigLoader.createDefault(file);
            const config = ConfigLoader.getDefault();
            new Logger("Config", config.logLevel).warn("Config does not exist, creating default '%s'", file);
            return config;
        }
        const fd: FileHandle = await open(file, "r");
        const data: string = await fd.readFile("utf-8");
        fd.close();

        return JSON.parse(data) as Config;
```

no need for `config` variable imo

I moved logger to the `if` and used the log level from default config (since that's the effective config)

Reply:
hmm lgtm, the config variable was there since there was a few usages of config#logLevel